constraints/src/main/java/gov/nasa/jpl/aerie/constraints/model/LinearEquation.java

```diff
@@ -44,14 +44,18 @@ public boolean changing() {
 
   private Optional<Duration> intersectionPointWith(final LinearEquation other) {
     if (this.rate == other.rate) return Optional.empty();
-    return Optional.of(
-        this.initialTime.plus(
-            Duration.roundNearest(
-                (other.valueAt(this.initialTime) - this.initialValue) / (this.rate - other.rate),
-                Duration.SECONDS
-            )
-        )
-    );
+    try {
+      return Optional.of(
+          this.initialTime.plus(
+              Duration.roundNearest(
+                  (other.valueAt(this.initialTime) - this.initialValue) / (this.rate - other.rate),
+                  Duration.SECONDS
+              )
+          )
+      );
+    } catch (ArithmeticException e) {
```

Comment:
I think this is an adequate guard :+1: My only concern is whether or not it's over-broad - but after inspecting `valueAt`, I've convinced myself that the only source of `ArithmeticException` should be the attempt to fit a large (or infinite) `double` into a `long`.

Reply:
Actually that's not true, I just yesterday saw a different long overflow from `.valueAt` because it attempted to find the duration of an interval like `[-MIN, 1s]`. This is fixed in the accumulated duration operation PR.

Reply:
Point being that computer arithmetic always finds a way.

Reply:
How about this: if we call `i = a.intersectionPointWith(b)` where `abs(i - a.initTime) > LONG_MAX` then we will get an overflow. This can happen if `a.initTime = Duration.MIN` and `i` is some normal positive value.

Its a bit pathological, but technically there's still a way.

Reply:
I've updated the check. It is no longer a try/catch. Instead it just checks this one case, just before the longs get involved.

There are still edge cases that can cause exceptions, like the above example, and I think we just let those exceptions happen. If we somehow end up with an extremely long interval or an extremely steep slope in a real simulation, something really has gone wrong and returning `Optional.empty()`, meaning there is no intersection` is probably incorrect.

Reply:
I tend to build a linear queue of expensive features I want to implement, which I know is not ideal, but here it is:

- In order to do a proper job of eliminating `Duration.MIN` and `Duration.MAX` from constraints/scheduling, my preference would be to use the lazy streaming profiles refactor idea, because that had very efficient and clean bounded evaluation.
- That refactor had major (major) problems with the parser system, that AFAICT made the objects behind interface that was simplest to use impossible to deserialize.
- Now that procedural scheduling/constraints is on the table, there is a long-term possibility of obviating the need for the scheduler/constraints parsers, enabling the lazy streaming interface again.

Therefore, in order to eliminate a couple idiosyncratic edge cases that could cause overflow exceptions, we should implement procedural constraints. I see no flaws in this logic.